File: problems/525/solution.py
class Solution:
    def findMaxLength(self, nums: List[int]) -> int:
        idx = defaultdict(list)
        idx[0] = [-1]
        cnt = 0
        
        for i in range(len(nums)):
            cnt += 1 if nums[i] else -1
            idx[cnt] += [i]
        
        return max(v[-1] - v[0] for v in idx.values())

File: problems/525/solution.go
package main
func findMaxLength(nums []int) int {
    m, count, ml := map[int]int{0:-1}, 0, 0
    // m[0] = -1
    
    for i := 0; i < len(nums); i++ {
        if nums[i] == 1 {
            count++
        } else {
            count--
        }
        
        if val, ok := m[count]; ok {
            ml = max(ml, i - val)
        } else {
            m[count] = i
        }
    }
    
    return ml
}

func max(a, b int) int {
    if a > b {
        return a
    }
    
    return b
}